### PR TITLE
Bugfix/restclient

### PR DESCRIPTION
**Pull request by stebenoist**
## Description of the change

- Add a pool of connections for Rest client

- Refactor in order to replace Unirest client by Spring Rest Template

- Add some useful debug logs

### CODE REVIEW

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java [hunk 1]
```diff
@@ -114,13 +92,13 @@ public void setProviderConfiguration(ProviderConfig providerConfiguration) throw
             SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(
                     sslContext,
                     NoopHostnameVerifier.INSTANCE);
-            CloseableHttpClient httpClient = HttpClients
+            httpClient = HttpClients
                     .custom()
```
Comment (laurentganne) [inline alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java]: To be able to deal with proxies defined in system properties, would need to replace **HttpClients.custom()** by
**HttpClientBuilder.create().useSystemProperties()**

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java [hunk 2]
```diff
@@ -131,85 +109,136 @@ public void setProviderConfiguration(ProviderConfig providerConfiguration) throw
...
+                    .setConnectionManager(poolHttpConnManager)
                     .setDefaultRequestConfig(clientConfig)
                     .setSslcontext(sslContext)
                     .build();
-            Unirest.setHttpClient(httpClient);
         } else {
-            CloseableHttpClient httpClient = HttpClients
+            httpClient = HttpClients
                     .custom()
```
Comment (laurentganne) [inline alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java]: Could replace HttpClients.custom() by HttpClientBuilder.create().useSystemProperties() to use proxies defined in system properties

File: alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java [hunk 3]
```diff
@@ -131,85 +109,136 @@ public void setProviderConfiguration(ProviderConfig providerConfiguration) throw
             }
 
             SSLContext sslContext = SSLContexts.createSystemDefault();
-
-            CloseableHttpClient httpClient = HttpClients
+            httpClient = HttpClients
                     .custom()
```
Comment (laurentganne) [inline alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java]: Could replace HttpClients.custom() by HttpClientBuilder.create().useSystemProperties() to use proxies defined in system properties

Comment (stebenoist) [reply alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java]: Yes ! good remark

Comment (stebenoist) [reply alien4cloud-yorc-plugin/src/main/java/org/ystia/yorc/alien4cloud/plugin/rest/RestClient.java]: I've just pushed a new commit on this

**Pull request merged**